src/renderer/components/dock/dock-tabs.tsx

```diff
@@ -93,17 +100,18 @@ export const DockTabs = ({ tabs, autoFocus, selectedTab, onChangeTab }: Props) =
 
 
   useEffect(() => {
-    if(!elem || !contentElem) return;
-
-    setContentWidth(contentElem.current.clientWidth);
-    setContainerWidth(elem.current.clientWidth);
-    setScrollPosition(elem.current.scrollLeft);
+    updateStateValues();
 
     // update values in store on scroll
     elem.current.addEventListener("scroll", updateScrollPosition);
 
     // update current values on resize to show/hide scroll
     window.addEventListener("resize", onWindowResize);
+
+    // update scroll state if tabs numbers has changed
+    dockStore.onTabsNumberChange(() => {
```

Comment:
I guess `useEffect` should return cleanup function for unsubscribing from scroll/resize/tabs-change events.